pandas/core/groupby/generic.py

```diff
@@ -358,9 +358,26 @@ def _cython_agg_general(
             if numeric_only and not is_numeric:
                 continue
 
-            result = self.grouper._cython_operation(
-                "aggregate", obj._values, how, axis=0, min_count=min_count
-            )
+            objvals = obj._values
+
+            if isinstance(objvals, Categorical):
```

Comment:
Does it make sense to try to separate this out further upstream in the future, so that _cython_agg_general isn't calling pure-python methods?

Reply:
Ideally we'll get this case handled within WrappedCythonOp._ea_wrap_cython_operation before too long, but im still troubleshooting that